lib/bolt/outputter/human.rb

```diff
@@ -104,10 +104,10 @@ def print_task_info(task)
           task['parameters'].each do |k, v|
             pretty_params << "- #{k}: #{v['type']}\n"
             pretty_params << "    #{v['description']}\n" if v['description']
-            usage << if !v['type'].to_s.include? "Optional"
-                       " #{k}=<value>"
-                     else
+            usage << if v['type'].to_s.include? "Optional"
```

Comment:
Is there a better way of testing for optional parameters?

Reply:
Using the compiler to test whether the undef is assignable to the type is more accurate than looking for Optional probably

Reply:
`Optional` is a little different for plans than for tasks. `Optional` means you can pass `undef` as a value, not that you don't need to provide a value. This is different from tasks, where we treat not providing a value and providing null as equivalent.

For this plan:

```puppet
plan foo(Optional[String] $param) {}
```

Trying to run it as `bolt plan run foo` will fail with `foo: expects a value for parameter 'param'`.

A parameter is *actually* optional if it has a default value.

```puppet
plan foo(String $param = 'foo') {}
```

This plan will run successfully with `bolt plan run foo`, since every parameter has a value.

Reply:
Oh I just saw that this is for tasks and not for plans. So kindly pretend I left this comment below where it's actually relevant. :)